sql/create.go

```diff
@@ -126,7 +126,28 @@ func (p *planner) CreateTable(n *parser.CreateTable) (planNode, error) {
 	// Inherit permissions from the database descriptor.
 	desc.Privileges = dbDesc.GetPrivileges()
 
-	if err := desc.AllocateIDs(); err != nil {
+	err = desc.AllocateIDs()
+	if err == errMissingPrimaryKey {
```

Comment:
instead of relying on this error, can you check for the presence of a primary key directly?